File: Scala/src/main/scala/composite/MenuItem.scala
package composite

case class MenuItem(name: String, description: String, vegetarian: Boolean, price: Double) extends MenuComponent {

  override def getName: String = name

  override def getDescription: String = description

  override def getPrice: Double = price

  override def isVegetarian: Boolean = vegetarian

  override def printMenu(): Unit = {
    print("  " + getName)
    if (isVegetarian) print("(v)")
    println(", " + getPrice)
    println("    -- " + getDescription)
  }

}

File: Kotlin/src/main/kotlin/chapter9_iterator/MenuItem.kt
package chapter9_iterator

import java.math.BigDecimal

data class MenuItem(
    private val name: String,
    private val description: String,
    private val vegetarian: Boolean,
    private val price: BigDecimal,
) {
    override fun toString(): String {
        return "$name, $price -- $description"
    }
}